sdk/rust/src/key/public_key/mod.rs

```diff
@@ -136,9 +142,36 @@ impl PublicKey {
         Err(Error::key_parse(format!("unsupported key algorithm: {}", info.algorithm.oid)))
     }
 
-    /// Return this private key, serialized as bytes.
+    pub fn from_str_der(s: &str) -> crate::Result<Self> {
+        Self::from_bytes_der(
+            &hex::decode(s.strip_prefix("0x").unwrap_or(s)).map_err(Error::key_parse)?,
+        )
+    }
+
+    pub fn from_str_ed25519(s: &str) -> crate::Result<Self> {
+        Self::from_bytes_ed25519(
+            &hex::decode(s.strip_prefix("0x").unwrap_or(s)).map_err(Error::key_parse)?,
+        )
+    }
+
+    pub fn from_str_ecdsa_secp256k1(s: &str) -> crate::Result<Self> {
```

Comment:
`*ecdsa_secp256k1*` appears throughout, but in the java SDK these are just called `ecdsa`, not sure about other SDKs.  Do we require these names to match?